src/hotspot/cpu/aarch64/javaFrameAnchor_aarch64.hpp

```diff
@@ -39,9 +39,11 @@
   //  3 - restoring an old state (javaCalls)
 
   void clear(void) {
+    // No fencing required, the members are declared volatile so the compiler will not reorder and
+    // the profiler always reads from the same thread and should observe the state in program order.
+
     // clearing _last_Java_sp must be first
     _last_Java_sp = nullptr;
-    OrderAccess::release();
```

Comment:
You still need a compiler fence here.

Reply:
Maybe not, if this is the same thread as readers. I don't know what it's for.

Reply:
`OrderAccess::release()` is more than a compiler barrier on AArch64. I am guessing this was copied from x86 or one of the others where that same function is a compiler barrier? Should I move `compiler_barrier` from the random source files its in, into globalDefinitions, and use that?

Reply:
As far as I can see, the frame anchor is only accessed from its owner thread, so it doesn't matter.

Reply:
I refactored `compiler_barrier()` into globalDefinitions and used it, it is potentially need to ensure the compiler doesn't change the store/load order so that `sp` is always last.